Source/NSDrawer.m

```diff
@@ -590,6 +598,46 @@ - (void) dealloc
 }
 
 // Opening and Closing
+- (NSRectEdge) _computeEdge
+{
+  NSRectEdge result = _preferredEdge;
+  NSRect windowFrame = [[self parentWindow] frame];
+  NSRect screenRect = [[NSScreen mainScreen] frame];
+  
+  switch (_preferredEdge)
+    {
+    case NSMinXEdge:
+      if (windowFrame.origin.x < _maxContentSize.width)
+	{
+	  result = NSMaxXEdge;
+	}
+      break;
+      
+    case NSMinYEdge:
+      if (windowFrame.origin.y < _maxContentSize.height)
+	{
+	  result = NSMaxYEdge;
+	}
+      break;
+      
+    case NSMaxXEdge:
+      if (windowFrame.origin.x + windowFrame.size.height > screenRect.size.height)
```

Comment:
Here the _maxContentSize.width isn't taken into consideration and both the height values should be width.